addon/helpers/shuffle.js

```diff
@@ -0,0 +1,39 @@
+import {
+  A as emberArray,
+  isEmberArray as isArray
+} from 'ember-array/utils';
+import Helper from 'ember-helper';
+import observer from 'ember-metal/observer';
+import set from 'ember-metal/set';
+import { typeOf } from 'ember-utils';
+
+export function shuffle(array, randomizer) {
+  let count = array.length;
+  let rand, temp;
+  randomizer = (typeOf(randomizer) === 'function' && randomizer) || Math.random;
+  
+  while (count > 1) {
+    rand = Math.floor(randomizer() * (count--));
+    
+    temp = array[count];
+    array[count] = array[rand];
+    array[rand] = temp;
+  }
+  
+  return emberArray(array);
```

Comment:
Could we return a new array? I would prefer to not mutate the `array` argument being passed in.